src/main/java/org/broadinstitute/hellbender/tools/walkers/contamination/ContaminationModel.java

```diff
@@ -160,15 +162,49 @@ public ContaminationModel(List<PileupSummary> sites) {
                 .mapToDouble(ps -> ps.getTotalCount() * oppositeAlleleFrequency.applyAsDouble(ps))
                 .sum();
 
-        final double contamination = contaminationOppositeDepth / totalDepthWeightedByOppositeFrequency;
+        final double contaminationEstimate = contaminationOppositeDepth / totalDepthWeightedByOppositeFrequency;
+
+        final double coeff1 = homs.stream().mapToDouble(ps -> oppositeAlleleFrequency.applyAsDouble(ps) * ps.getTotalCount()).sum();
+        final double coeff2 = homs.stream().mapToDouble(ps ->
+                oppositeAlleleFrequency.applyAsDouble(ps)*(1 - oppositeAlleleFrequency.applyAsDouble(ps)) * MathUtils.square(ps.getTotalCount())
+        ).sum();
+
+        final DoubleUnaryOperator errorFunc = c -> homs.isEmpty() ? 1 : Math.sqrt(coeff1*c*(1-c) + coeff2*c*c) / totalDepthWeightedByOppositeFrequency;
+
+        // we're going to binary search to find the largest contamination whose expected standard error brings it within range of
+        // our estimate.  That is, suppose we estimate a contamination of 0.03 and the standard error of 0.05 is 0.02.  Then 0.05 is
+        // the upper end of our 1-sigma confidence interval.
+        // this binary search is far from optimized (in fact we could solve this explicitly with a messy closed-form solution)
+        // but it converges to a precision of 1e-6 in 20 iterations of the square root of a linear function.  This is fast enough.
+        double top = 1.0;
```

Comment:
Could you either find a standard binary search, or generalize this in MathUtils?

If you add it in MathUtils, create a simple test for it.

Reply:
done